Project/sources/methods.py

```diff
@@ -37,22 +36,27 @@ def DetectBarcode(img):
     Detect barcode in the frame
     """
     gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+    detector = cv2.barcode_BarcodeDetector()
 
-    for barcode in decode(gray):
-        isbn = barcode.data.decode('utf-8')
-        
-        isbn_meta = ParseISBN(isbn)
+    # Detecting the barcode
+    valid, decoded_info, decoded_type, corners = detector.detectAndDecode(gray)
```

Comment:
These variables need to be documented.

source: https://opencv.org/recognizing-one-dimensional-barcode-using-opencv